tests/integration/golang/mlflow/namespace/flows/experiment_test.go

```diff
@@ -0,0 +1,598 @@
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageActive),
+			},
+		},
+	)
+	s.getExperimentByIDAndCompare(
+		namespace2Code,
+		experiment2ID,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:   experiment2ID,
+				Name: "UpdatedExperiment2",
+				Tags: []response.ExperimentTagPartialResponse{
+					{
+						Key:   "KeyTag2",
+						Value: "ValueTag2",
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageActive),
+			},
+		},
+	)
+
+	// 8. test `POST /experiments/delete` endpoint.
+	s.deleteExperiment(namespace1Code, experiment1.Experiment.ID)
+	s.deleteExperiment(namespace2Code, experiment2.Experiment.ID)
+
+	// check that experiment lifecycle has been updated.
+	s.getExperimentByIDAndCompare(
+		namespace1Code,
+		experiment1ID,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:   experiment1ID,
+				Name: "UpdatedExperiment1",
+				Tags: []response.ExperimentTagPartialResponse{
+					{
+						Key:   "KeyTag1",
+						Value: "ValueTag1",
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageDeleted),
+			},
+		},
+	)
+	s.getExperimentByIDAndCompare(
+		namespace2Code,
+		experiment2ID,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:   experiment2ID,
+				Name: "UpdatedExperiment2",
+				Tags: []response.ExperimentTagPartialResponse{
+					{
+						Key:   "KeyTag2",
+						Value: "ValueTag2",
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageDeleted),
+			},
+		},
+	)
+
+	// 9. test `POST /experiments/restore` endpoint.
+	s.restoreExperiment(namespace1Code, experiment1ID)
+	s.restoreExperiment(namespace2Code, experiment2ID)
+
+	// check that experiment lifecycle has been updated.
+	s.getExperimentByIDAndCompare(
+		namespace1Code,
+		experiment1ID,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:   experiment1ID,
+				Name: "UpdatedExperiment1",
+				Tags: []response.ExperimentTagPartialResponse{
+					{
+						Key:   "KeyTag1",
+						Value: "ValueTag1",
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageActive),
+			},
+		},
+	)
+	s.getExperimentByIDAndCompare(
+		namespace2Code,
+		experiment2ID,
+		&response.GetExperimentResponse{
+			Experiment: &response.ExperimentPartialResponse{
+				ID:   experiment2ID,
+				Name: "UpdatedExperiment2",
+				Tags: []response.ExperimentTagPartialResponse{
+					{
+						Key:   "KeyTag2",
+						Value: "ValueTag2",
+					},
+				},
+				ArtifactLocation: "/artifact/location",
+				LifecycleStage:   string(models.LifecycleStageActive),
+			},
+		},
+	)
+}
+
+func (s *ExperimentFlowTestSuite) createExperiment(
```

Comment:
It was not that clear to me that all these support methods for create/update etc were also performing assertions - can we name these in a way that makes that clear?

Reply:
it you check, all the methods where we use `assertion` have a prefix `Compare`